src/Microsoft.Azure.SignalR.Serverless.Protocols/MessagePackServerlessProtocol.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.IO;
+
+using MessagePack;
+
+namespace Microsoft.Azure.SignalR.Serverless.Protocols
+{
+    public class MessagePackServerlessProtocol : IServerlessProtocol
+    {
+        public int Version => 1;
+
+        public bool TryParseMessage(ref ReadOnlySequence<byte> input, out ServerlessMessage message)
+        {
+            var array = input.ToArray();
+            var startOffset = 0;
+            _ = MessagePackBinary.ReadArrayHeader(array, startOffset, out var readSize);
+            startOffset += readSize;
+            var messageType = MessagePackHelper.ReadInt32(array, ref startOffset, "messageType");
+            switch (messageType)
+            {
+                case ServerlessProtocolConstants.InvocationMessageType:
+                    message = ConvertInvocationMessage(array, ref startOffset);
+                    break;
+                default:
+                    // TODO:OpenConnectionMessage and CloseConnectionMessage only will be sent in JSON format. It can be added later.
+                    message = null;
+                    break;
+            }
+
+            return message != null;
+        }
+
+        private static InvocationMessage ConvertInvocationMessage(byte[] input, ref int offset)
```

Comment:
>ConvertInvocationMessage [](start = 41, length = 24)

Seems same as the method in MessagePackHelper. Can be merged?

Reply:
Oh, this should be removed in helper in `MessagePackHelper`.